maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/JarMojo.java

```diff
@@ -173,6 +219,53 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         project.getArtifact().setFile(jarFile);
     }
 
+    private void validateArchetypeScripts(Map<String, List<String>> resources) throws MojoExecutionException {
+        File schemaFile = getArchetypeSchemaFile();
+        if (schemaFile == null) {
+            return;
+        }
+        try {
+            SchemaFactory factory = SchemaFactory.newInstance(SCHEMA_LANG);
+            Schema schema = factory.newSchema(schemaFile);
+            Validator validator = schema.newValidator();
+
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            DocumentBuilder db = dbf.newDocumentBuilder();
+
+            for (Entry<String, List<String>> resourcesEntry : resources.entrySet()) {
+                for (String resource : resourcesEntry.getValue()) {
+                    if (FilenameUtils.getExtension(resource).equalsIgnoreCase("xml")){
+                        File xmlFile = Path.of(resourcesEntry.getKey(), resource).toFile();
+                        Document doc = db.parse(xmlFile);
+                        if (doc.getDocumentElement().getNodeName().equals(ARCHETYPE_ROOT_ELEMENT)) {
+                            validator.validate(new StreamSource(xmlFile));
+                        }
+                    }
+                }
+            }
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private File getArchetypeSchemaFile() {
```

Comment:
We may be able to skip the use of aether completely. Try adding the schema as a dependency in the plugin pom.xml and then use the class-loader to get the resource input stream.